client/my-sites/plugins/with-plugin-redirect.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { isJetpackSite, getSiteWoocommerceWizardUrl } from 'calypso/state/sites/selectors';
+import { isPluginActionCompleted } from 'calypso/state/plugins/installed/selectors';
+import { INSTALL_PLUGIN } from 'calypso/lib/plugins/constants';
+import getSiteConnectionStatus from 'calypso/state/selectors/get-site-connection-status';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import { successNotice, removeNotice } from 'calypso/state/notices/actions';
+
+// Plugins list that should perfomr a redirect
+// once it's installed.
+const redirectingPluginsList = {
+	woocommerce: getSiteWoocommerceWizardUrl,
+};
+
+/**
+ * Return the handler function to get the plugin redirect URL.
+ *
+ * @param {string} slug - Plugin slug.
+ * @returns {Function|boolean} Selector handler if the plugin should redirect. Otherwise, False.
+ */
+function getPluginRedirectUrl( slug ) {
+	if ( Object.keys( redirectingPluginsList ).indexOf( slug ) < 0 ) {
```

Comment:
I think we can simplify this with `Object.prototype.hasOwnProperty`:

```suggestion
	if ( redirectingPluginsList.hasOwnProperty( slug ) ) {
```

Reply:
<3